plugins/nexus-repository-pypi/src/main/java/org/sonatype/nexus/repository/pypi/internal/PyPiLinkInfo.java

```diff
@@ -0,0 +1,56 @@
+package org.sonatype.nexus.repository.pypi.internal;
+
+import javax.annotation.Nonnull;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Container for URL PEP 503 compatible urls.
+ */
+final class PyPiLinkInfo implements Comparable<PyPiLinkInfo> {
+    private final String file;
+
+    private final String link;
+
+    private final String dataPythonRequires;
+
+    PyPiLinkInfo(@Nonnull final String file, @Nonnull final String link, @Nonnull final String dataPythonRequires) {
+        this.file = checkNotNull(file);
+        this.link = checkNotNull(link);
```

Comment:
The link is not always built the same. It depend if it's an index link or a root index link. This is done in `PyPiIndexUtils.indexLinkToMap` and `PyPiIndexUtils.rootIndexLinkToMap`. This make me think that I can simplify things and ensure the "/" is added in the build phase. 

This also make me think that since there's two kind of link and two way of building them, maybe I should use two classes.